### PR TITLE
Add Aerospike handler

### PR DESCRIPTION
**Pull request by geobeau**
Next step:
- Explore the get policies and see if we need to have some part configurable to tune
- Make a scalability test

### CODE REVIEW

File: app/memcached_cluster_proxy.go [hunk 1]
```diff
@@ -124,12 +128,28 @@ func main() {
...
+	case "L1":
+		orca = orcas.L1Only
+		break
+	default:
+		orca = orcas.L1OnlyForwardGet
+		break
+	}
+
 	if dstType == "noop" {
 		log.Printf("Starting Rend (in GET forwarder mode) on port %d", listenPort)
```
Comment (ashangit) [inline app/memcached_cluster_proxy.go]: Not true anymore

File: app/memcached_cluster_proxy.go [hunk 2]
```diff
@@ -124,12 +128,28 @@ func main() {
...
-		go server.ListenAndServe(l, protocols, server.Default, orcas.L1OnlyForwardGet, sourceCluster, backfillCluster)
+		go server.ListenAndServe(l, protocols, server.Default, orca, sourceCluster, backfillCluster)
 	} else {
 		log.Printf("Starting Rend (backfill mode) on port %d", listenPort)
-		go server.ListenAndServe(l, protocols, server.Default, orcas.Backfill, sourceCluster, backfillCluster)
+		go server.ListenAndServe(l, protocols, server.Default, orca, sourceCluster, backfillCluster)
```
Comment (ashangit) [inline app/memcached_cluster_proxy.go]: Don't we need to enforce backfill if dstType define

File: app/memcached_cluster_proxy.go [hunk 3]
```diff
@@ -124,12 +128,28 @@ func main() {
 	sourceCluster := newHandlerFromConfig(srcType, srcHostnames, srcClusterName, srcClusterDC, srcBucketName)
 	backfillCluster := newHandlerFromConfig(dstType, dstHostnames, dstClusterName, dstClusterDC, dstBucketName)
 
+	var orca func(l1 handlers.Handler, l2 handlers.Handler, res protocol.Responder) orcas.Orca
+	switch queryLayout {
```
Comment (ashangit) [inline app/memcached_cluster_proxy.go]: Do we rely need all of these?
Seems to me that the sdk is already managing this and that we only need L1Only

File: handlers/aerospike/aerospike.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+	}
+}
+
+func (h Handler) Close() error {
+	h.client.Close()
+	return nil
+}
+
+func (h Handler) Set(cmd common.SetRequest) error {
+	key, err := aero.NewKey("ssd", "rend", cmd.Key)
```
Comment (ashangit) [inline handlers/aerospike/aerospike.go]: we should not name the namespace ssd or nvme (perhaps something like persisted?)

File: handlers/aerospike/aerospike.go [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+	log.Println("Key: ", key, err, cmd.Exptime)
+	if err != nil {
+		return err
+	}
+	data := aero.BinMap{"bin1": cmd.Data}
```
Comment (ashangit) [inline handlers/aerospike/aerospike.go]: bin1 -> value?

**Review by ashangit — CHANGES_REQUESTED** [review]

File: handlers/aerospike/aerospike.go [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+	port, err := strconv.Atoi(addr[1])
+	if err != nil {
+		return Handler{}, err
+	}
+	client, err := aero.NewClient(addr[0], port)
+	if err != nil {
+		return Handler{}, err
+	}
+
+	return Handler{client: client, namespace: bucketName, set: "rend"}, nil
```
Comment (ashangit) [inline handlers/aerospike/aerospike.go:34]: I think that the bucket notion is more the set on aerospike

Comment (geobeau) [reply handlers/aerospike/aerospike.go:34]: Yes but it was a cheap way to be able adapt the namespace for testing (so I can change between ssd/nvme/persistent for my tests)

Comment (ashangit) [reply handlers/aerospike/aerospike.go:34]: ok

**Review by ashangit — APPROVED** [review]

**Pull request merged**